### DISCUSSION TITLE
Ad a setting to allow regexp search for parts

### DISCUSSION THREAD

**Discussion by SergeoLacruz** [inventree/InvenTree#4265] [Q&A]
Hi,
I tried to enable regular expression search in the part search. It is easy to add an additional setting to Inventree.
But I get stuck in parts/api.py. The search fields are defind here and I know I need to add a $ sign to the description.
The code fragment below works but requires a server restart because the definition is not updated when I just change the setting. 
Any hint from the experts?

```python
class PartList(APIDownloadMixin, ListCreateAPI):
...

    search_fields = [
        'name',
        InvenTreeSetting.PartSearchDescription[InvenTreeSetting.get_setting('SEARCH_REGEX', False)],
#        'description',
        'IPN',
        'revision',
        'keywords',
        'category__name',
        'manufacturer_parts__MPN',
        'supplier_parts__SKU',
    ]
```

Michael

**Comment by SchrodingersGat** [comment]
The `PartList` class attribute *search_fields* is evaluated when the code is first loaded, so it will (as written here) require a restart of the server to function as expected.

There is some guidelines on how to make these search fields behave dynamically in the [DRF documentation](https://www.django-rest-framework.org/api-guide/filtering/#searchfilter)
